### PR TITLE
feat(sensor): formatting options and the history graph

### PR DESCRIPTION
**Pull request by fx**
## Summary

First PR of `docs/changes/0018-sensor-cards-to-spec.md`: the sensor card's formatting options and its history graph. `displayPrecision`, `unitOverride` and `valueScale` layer onto the shipped device-class matrix through one ordered pipeline shared by the value, the trend delta and the min/max footer; `showGraph` / `graphHours` / `graphMode` render the `liebe-spark` anatomy across `row`, `tall` and `full`; `showTrend` puts the window's movement beside the `glance` value. Binary sensor is PR 2; spec sync is PR 3.

## The existing formatting matrix was pinned before it was changed

The first commit on this branch adds tests for the current device-class formatting behaviour and changes nothing. The options then layer on top, so the tests prove the new pipeline preserves the old matrix rather than replacing it — a formatting regression here would be silent and would affect every sensor on every dashboard.

## Aggregation mode is chosen per rendering surface

`sample` for line graphs and the footer's extremes, `delta` for bars and the trend arrow. So a default line graph on a counter shows its cumulative curve, while a `full`-tier bar card takes both projections of a single cached window — the projection cache from change 0015 is keyed on entity + window + mode + points, so this costs one fetch rather than two.

## Graph availability is the pipeline's judgement, never the card's

The hook resolves a text sensor `unsupported` from its live state before it costs a request. `unsupported`, error and an empty window all degrade to the graph-less layout; only *loading* renders a box with no data in it, holding the graph's final height so nothing moves when the series lands. Re-deriving "is this graphable" in the card would give two answers that drift.

## Input shapes with no branch behind them

After the branch-coverage check, the separate question — what shapes can these inputs legally take, and does the code inspect them — was run against the consumers rather than the card. A sensor `state` may be `unknown`, `unavailable`, empty or non-numeric; `unit_of_measurement` may be absent; and a history window may be empty or hold a single point, which is what a min/max footer over one sample and a spark over zero samples turn on. `displayPrecision`, `valueScale` and `graphHours` arrive from card configuration, and by the dashboard-config forward-compatibility rule a junk value gets read rather than rejected, so each is normalised at the boundary with `NaN`, `Infinity`, negative, zero and fractional cases tested.

## Testing

- [x] `npm test` — 2431 passed, 2 skipped
- [x] `npm run lint`, `npm run typecheck`
- [x] `npm run test:coverage` — patch clean, verified against `BRDA` branch entries rather than the percentage summary
- [x] `npm run build-storybook` and `npm run build:ha:prod`

Stories cover the loading, empty and unsupported states as distinct cases, and assert rather than merely rendering.

### CODE REVIEW

File: src/store/sensorOptions.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+ * Clamp a stored window to one the card can actually draw.
+ *
+ * `graphHours` arrives from card configuration, so a junk value gets read
+ * rather than rejected, and every junk shape means the same thing: the document
+ * does not describe a window, so the default one applies. `NaN`, `Infinity`,
+ * `0`, a negative number and a non-number all land there; a real number outside
+ * the option doc's bounds is a window that was described badly rather than not
+ * at all, and clamps. Fractions survive the clamp — the pipeline's own
+ * normaliser keeps them, and half an hour is a legitimate ask.
+ *
```
[the remark this answers is not in the document]
Comment (fx) [reply src/store/sensorOptions.ts]: Fixed in c3698ac. The comment now states what the code does: `MIN_SENSOR_GRAPH_HOURS` is 1, so a sub-hour window clamps up rather than surviving, and fractions above the minimum are honoured as written (1.5 stays 1.5). The sensor card is deliberately stricter than `normalizeHistoryHours` underneath it, which keeps fractions down to any positive value — the reason is recorded rather than left for a reader to assume the two agree.

The test changed shape as well as content: the boundary cases were asserting against `MIN_SENSOR_GRAPH_HOURS` and `MAX_SENSOR_GRAPH_HOURS`, which means they would have agreed with the code whatever those constants became — including a change to 0.5, which would have made the sub-hour case pass while silently widening the option's contract. They now assert literals, with a separate case pinning the constants themselves.
